numpy/core/arrayprint.py

```diff
@@ -51,24 +51,35 @@
     _MAXINT = sys.maxint
     _MININT = -sys.maxint - 1
 
-def product(x, y):
-    return x*y
+_format_options = {
+    'edgeitems': 3,  # repr N leading and trailing items of each dimension
+    'threshold': 1000,  # total items > triggers array summarization
+    'precision': 8,
+    'suppress': False,
+    'linewidth': 75,
+    'nanstr': 'nan',
+    'infstr': 'inf',
+    'sign': '-',
+    'formatter': None }
```

Comment:
This overlaps with the changes in #9605. Which should we prefer?

Reply:
I think I prefer the use of `dict(edgeitems=)` in that implementation, but `_make_options_dict` is nice in this one

Reply:
NP @eric-wieser.  I am closing the PR.  😄